### DISCUSSION TITLE
Basic Auth

### DISCUSSION THREAD

**Discussion by Teicu** [projectdiscovery/nuclei#1608] [Q&A] [answered]
Hi Hello 

I want to scan a site that is protected by Basic Auth, does this scanner support this type of authentication? And if you could show me for example the command I should use.

Thank you

**Comment by ehsandeep** [comment, marked as answer]
@Teicu you can include basic auth header token with all the request by including custom header using CLI flag, for example:-

```
nuclei -u https://target.com -H "Authorization: Basic QWxhZGRpbjpvcGVuIHNlc2FtZQ=="
```

`QWxhZGRpbjpvcGVuIHNlc2FtZQ==` is an example value that you can update on your side.

**Reply by peterbenjamin-kh** [reply]
In case of:
- `-list hosts.txt` where each URL require different Authorization headers, is there way to do this?
- `-list hosts.txt` where all URLs expect the same Authorization header, would this work? `nuclei -l hosts.txt -H "Authorization: Basic QWxhZGRpbjpvcGVuIHNlc2FtZQ=="` 

Apologies if this is documented somewhere, I searched through nuclei docs and couldn't find relevant info.

**Comment by Teicu** [comment]
Hi, Thank you very much for your answer. It works like a charm! 

Cheers
